src/intTest/java/uk/nhs/digital/nhsconnect/lab/results/inbound/InboundUserAcceptanceTest.java

```diff
@@ -48,14 +50,33 @@ void tearDown() {
         System.setProperty("LAB_RESULTS_SCHEDULER_ENABLED", "false");
     }
 
-    @Test
-    void testSendEdifactIsProcessedAndPushedToGpOutboundQueue()
-        throws JMSException, IOException, InterchangeParsingException, MessageParsingException, JSONException {
-
-        final String content = new String(Files.readAllBytes(getEdifactResource().getFile().toPath()));
-
-        final OutboundMeshMessage outboundMeshMessage = OutboundMeshMessage.create(RECIPIENT,
-            WorkflowId.PATHOLOGY, content, null);
+    /**
+     * The test EDIFACT message (.dat file) is sent to the MESH mailbox where the adaptor receives inbound messages.
+     * The test waits for the messages to be processed and compares the FHIR message published to the GP Outbound Queue
+     * with the expected FHIR representation of the original message sent (.json file having the same name as the .dat)
+     */
+    @ParameterizedTest(name = "[{index}] - {0}")
+    @ArgumentsSource(SuccessArgumentsProvider.class)
+    void testEdifactIsSuccessfullyProcessedAndPushedToGpOutboundQueue(String category, TestData testData)
+            throws JMSException, InterchangeParsingException, MessageParsingException, JSONException {
+
+        final String recipient = getEdifactParser().parse(testData.getEdifact())
+            .getInterchangeHeader().getRecipient();
+
+        WorkflowId workflowId;
+        if (testData.getEdifact().contains(WorkflowId.PATHOLOGY.getWorkflowId())) {
+            workflowId = WorkflowId.PATHOLOGY;
+        } else if (testData.getEdifact().contains(WorkflowId.SCREENING.getWorkflowId())) {
+            workflowId = WorkflowId.SCREENING;
+        } else {
+            throw new RuntimeException("Unsupported Workflow ID");
+        }
+
+        final boolean ackRequested = testData.getEdifact().replace("\n", "")
+            .matches(ACK_REQUESTED_REGEX);
```

Comment:
Seems to work fine, I'll remove that

Reply:
That could be handled by adjusting the regex:
```
    static final String ACK_REQUESTED_REGEX = "^.*UNB\\+UNOC.*\\+\\+1'\\s*UNH.*$";
```

Reply:
That regex doesn't seem to work, as far as I can tell its because .* can't match the newlines throughout the rest of the file.

I've adjusted it a bit more, this seems to be working:

`static final String ACK_REQUESTED_REGEX = "^[\\S\\s]*UNB\\+UNOC.*\\+\\+1'\\s*UNH[\\S\\s]*$";`

Reply:
Ah I see; probably the "correct" way then is to turn on the multiline flag:
```
    static final String ACK_REQUESTED_REGEX = "(?m)^.*UNB\\+UNOC.*\\+\\+1'\\s*UNH.*$";
```
But running the latest on your branch on my local env works fine - have you added/adjusted some of the test edifact+json pairs yet to be pushed?

Reply:
This works, and seems more correct. Updated the regex.